js/__internal/grids/grid_core/views/m_columns_view.ts

```diff
@@ -1008,121 +1024,101 @@ export class ColumnsView extends viewWithColumnStateMixin {
 
   }
 
-  _getWidths($cellElements) {
-    const result: any[] = [];
-    let width;
+  _getWidths($cellElements?: dxElementWrapper): number[] {
+    if (!$cellElements) {
+      return [];
+    }
 
-    if ($cellElements) {
-      iteratorUtils.each($cellElements, (index, item) => {
-        width = item.offsetWidth;
-        if (item.getBoundingClientRect) {
-          const clientRect = getBoundingRect(item);
-          if (clientRect.width > width - 1) {
-            width = clientRect.width;
-          }
-        }
+    const result: number[] = [];
+    const cellElements = $cellElements.toArray();
 
-        result.push(width);
-      });
-    }
+    (cellElements as HTMLElement[]).forEach((cell) => {
+      const rect = getBoundingRect(cell);
+      const width = rect.width > cell.offsetWidth - 1 ? rect.width : cell.offsetWidth;
+
+      result.push(width);
+    });
 
     return result;
   }
 
-  getColumnWidths($tableElement) {
-    const that = this;
-    let result: any[] = [];
-    let $rows;
-    let $cells;
-
+  getColumnWidths($tableElement?: dxElementWrapper): number[] {
     (this.option('forceApplyBindings') || noop)();
 
-    $tableElement = $tableElement || that.getTableElement();
+    $tableElement = $tableElement ?? this.getTableElement();
 
     if ($tableElement) {
-      $rows = $tableElement.children('tbody:not(.dx-header)').children();
+      const $rows = $tableElement.children('tbody:not(.dx-header)').children();
 
       for (let i = 0; i < $rows.length; i++) {
         const $row = $rows.eq(i);
-        const isRowVisible = $row.get(0).style.display !== 'none' && !$row.hasClass('dx-state-invisible');
-        if (!$row.is(`.${GROUP_ROW_CLASS}`) && !$row.is(`.${DETAIL_ROW_CLASS}`) && !$row.is(`.${ERROR_ROW_CLASS}`) && isRowVisible) {
-          $cells = $row.children('td');
-          break;
+
+        const isGroupRow = $row.hasClass(GROUP_ROW_CLASS);
+        const isDetailRow = $row.hasClass(DETAIL_ROW_CLASS);
+        const isErrorRow = $row.hasClass(ERROR_ROW_CLASS);
+
+        const isRowVisible = $row.is(':visible');
+        const isRelevantRow = !isGroupRow && !isDetailRow && !isErrorRow;
+
+        if (isRowVisible && isRelevantRow) {
+          const $cells = $row.children('td');
+
+          const result = this._getWidths($cells);
+
+          return result;
         }
       }
-
-      result = that._getWidths($cells);
     }
 
-    return result;
+    return [];
   }
 
   getVisibleColumnIndex(columnIndex, rowIndex) {
     return columnIndex;
   }
 
-  setColumnWidths({
-    widths, $tableElement, columns, fixed,
```

Comment:
No more need in $tableElement and fixed parameters, because $tableElement is set only in ColumnFixing.setColumnWidths, but I divided logic of this method into to methods. For usual table this method is called, for fixed columns setFixedTableColumnWidths is called